modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1130,25 +1138,33 @@ private CompletableFuture<Void> continueReadOnlyIndexLookup(
      */
     // TODO: need to properly handle primary replica changes https://issues.apache.org/jira/browse/IGNITE-17615
     private CompletableFuture<Void> processTxFinishAction(TxFinishReplicaRequest request, String txCoordinatorId) {
-        List<TablePartitionId> aggregatedGroupIds = request.groups().values().stream()
-                .flatMap(List::stream)
-                .map(IgniteBiTuple::get1)
-                .collect(toList());
+        Map<TablePartitionId, IgniteBiTuple<ClusterNode, Long>> partitionMap = toPartitionMap(request.groups());
```

Comment:
Why do we need this map? And why do we need Long term for cleanup requests.

- Do we check term (enlistmentConsistencyToken) for cleanup on the server side?
- If we do, do we really need this?

FYI We will cover mandatory lock release on lease expiration within dataNodes recovery track.

Reply:
I believe that for cleanup request we should skip this check. WDYT?

Reply:
> If the old one did not crash - it would still hold the locks.
We will cover it within dataNodes recovery track. On expiration, primary replica will release all locks.

> Am I missing something?
I don't think so. I also believe that we may get rid of term inTxCleanupReplicaRequest.
@vldpyatkov what do you think?